third_party/move/move-compiler-v2/src/pipeline/split_critical_edges_processor.rs

```diff
@@ -0,0 +1,376 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This pass splits critical edges with empty blocks.
+//! A critical edge is an edge where the source node has multiple successors,
+//! and the target node has multiple predecessors.
+//!
+//! Side effects: clear existing annotations.
+//!
+//! Prerequisites: no call instructions have abort actions.
+//!
+//! Postconditions: no critical edges in the control flow graph.
+
+use move_model::{ast::TempIndex, model::FunctionEnv};
+use move_stackless_bytecode::{
+    function_target::FunctionData,
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::{AttrId, Bytecode, Label},
+};
+use std::collections::{BTreeMap, BTreeSet};
+
+pub struct SplitCriticalEdgesProcessor {}
+
+impl FunctionTargetProcessor for SplitCriticalEdgesProcessor {
+    fn process(
+        &self,
+        _targets: &mut FunctionTargetsHolder,
+        fun_env: &FunctionEnv,
+        mut data: FunctionData,
+        _scc_opt: Option<&[FunctionEnv]>,
+    ) -> FunctionData {
+        if cfg!(debug_assertions) {
+            Self::check_precondition(&data);
+        }
+        if fun_env.is_native() {
+            return data;
+        }
+        let mut transformer = SplitCriticalEdgesTransformation::new(std::mem::take(&mut data.code));
+        transformer.transform();
+        data.code = transformer.code;
+        data.annotations.clear();
+        if cfg!(debug_assertions) {
```

Comment:
Please make this `if cfg!(debug_assertions) || log_enabled!(Level::Debug)`. Above too.

We may want to keep this kind of functionality even in optimized code, as long as it is only costs if enabled.

Reply:
done